staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -150,6 +152,15 @@ type SharedInformer interface {
 	// because the implementation takes time to do work and there may
 	// be competing load and scheduling noise.
 	AddEventHandlerWithResyncPeriod(handler ResourceEventHandler, resyncPeriod time.Duration)
+	// RemoveEventHandler removes a formerly added event handler, again.
+	// It is possible to remove only event handlers that are (go) comparable,
+	// This means event handlers like ResourceEventHandlerFuncs cannot be removed
+	// if added by value and not by reference, because they contain function pointers
+	// that are not comparable in go. Trying to remove an uncomparable handler
+	// results in an appropriate error.
+	// Please note: If, for some reason, the same handler has been added multiple
+	// times, all registrations will be removed.
+	RemoveEventHandler(handler ResourceEventHandler) error
```

Comment:
hmm ok, it looks like it will be changing back to returning an error, I will wait until https://github.com/kubernetes/kubernetes/pull/98657 is updated to sync with latest